src/components/side_panel/side_panel/side_panel.ts

```diff
@@ -152,37 +154,25 @@ css/* scss */ `
   }
 `;
 
-interface Props {
-  component: string;
-  panelProps: any;
-  onCloseSidePanel: () => void;
-}
-
-interface State {
-  panel: SidePanelContent;
-}
-
-export class SidePanel extends Component<Props, SpreadsheetChildEnv> {
+export class SidePanel extends Component<never, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-SidePanel";
-  static props = {
-    component: String,
-    panelProps: { type: Object, optional: true },
-    onCloseSidePanel: Function,
-  };
-  state!: State;
+  static props = {};
+  sidePanelStore!: Store<SidePanelStore>;
 
   setup() {
-    this.state = useState({
-      panel: sidePanelRegistry.get(this.props.component),
-    });
-    onWillUpdateProps(
-      (nextProps: Props) => (this.state.panel = sidePanelRegistry.get(nextProps.component))
-    );
+    this.sidePanelStore = useStore(SidePanelStore);
+  }
+
+  get panel() {
+    return sidePanelRegistry.get(this.sidePanelStore.component);
+  }
+
+  close() {
+    this.sidePanelStore.close();
   }
 
   getTitle() {
-    return typeof this.state.panel.title === "function"
-      ? this.state.panel.title(this.env)
-      : this.state.panel.title;
+    const panel = this.panel;
+    return typeof panel.title === "function" ? panel.title(this.env) : panel.title;
```

Comment:
same here, this coule be done in the store

Reply:
the store doesn't have access to the env (and should not). Funnily enough, I don't think we ever use the dynamic title. It could probably be removed